### DISCUSSION TITLE
Suggestion Regarding Document Retrieval in OpenWebUI

### DISCUSSION THREAD

**Discussion by BennisonDevadoss** [open-webui/open-webui#5857] [General]
I've been using OpenWebUI recently and encountered something that might be a bug, though I’m not entirely sure. Here's the situation:

When I upload multiple documents, the RAG (Retrieval-Augmented Generation) system retrieves the top 5 chunks from each document for answering queries. This raises a concern: since I’m uploading multiple documents, the combined context length increases significantly. While I can extend the context length to a certain limit, it's not infinite, meaning the model cannot fully understand the entire context.

Additionally, this seems to affect the accuracy of the citations. The chunks retrieved across all uploaded documents don’t always align closely with the user's query, making the citations less accurate in the WebUI interface.

To address this, I suggest that instead of creating separate collections for each uploaded document, all documents could be stored in a single collection. This way, when querying, the retrieval would occur against that single collection, returning a maximum of 5 chunks. This approach would better manage context length and improve citation accuracy.

What do you all think? Is this an actual issue, or am I missing something?

**Comment by tdiprima** [comment]
@BennisonDevadoss I'm experiencing something similar. I have over 1,000 short documents in a collection, but when I ask a question, only 3 documents get retrieved, even though more should be relevant. Like you mentioned, it feels like there’s a limit on how many documents RAG can "see" at once, which impacts the completeness of the response and accuracy of document listings.

Your idea of consolidating into a single collection sounds promising -- I had created an issue that they should bring back the "All Documents" search, but they basically said no. #6564 

Do you have any insights on configuring any retrieval parameters to ensure broader access to the collection?
